stable/concourse/templates/web-deployment.yaml

```diff
@@ -97,255 +86,813 @@ spec:
+            - name: CONCOURSE_LDAP_GROUP_SEARCH_USER_ATTR
+              value: {{ .Values.concourse.web.auth.ldap.groupSearchUserAttr }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.host }}
+            - name: CONCOURSE_LDAP_HOST
+              value: {{ .Values.concourse.web.auth.ldap.host }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.insecureNoSsl }}
+            - name: CONCOURSE_LDAP_INSECURE_NO_SSL
+              value: {{ .Values.concourse.web.auth.ldap.insecureNoSsl }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.insecureSkipVerify }}
+            - name: CONCOURSE_LDAP_INSECURE_SKIP_VERIFY
+              value: {{ .Values.concourse.web.auth.ldap.insecureSkipVerify }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.startTls }}
+            - name: CONCOURSE_LDAP_START_TLS
+              value: {{ .Values.concourse.web.auth.ldap.startTls }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchBaseDn }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_BASE_DN
+              value: {{ .Values.concourse.web.auth.ldap.userSearchBaseDn }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchEmailAttr }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_EMAIL_ATTR
+              value: {{ .Values.concourse.web.auth.ldap.userSearchEmailAttr }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchFilter }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_FILTER
+              value: {{ .Values.concourse.web.auth.ldap.userSearchFilter }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchIdAttr }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_ID_ATTR
+              value: {{ .Values.concourse.web.auth.ldap.userSearchIdAttr }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchNameAttr }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_NAME_ATTR
+              value: {{ .Values.concourse.web.auth.ldap.userSearchNameAttr }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchScope }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_SCOPE
+              value: {{ .Values.concourse.web.auth.ldap.userSearchScope }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.ldap.userSearchUsername }}
+            - name: CONCOURSE_LDAP_USER_SEARCH_USERNAME
+              value: {{ .Values.concourse.web.auth.ldap.userSearchUsername }}
+            {{- end }}
+            {{- end }}
+
+            {{- if .Values.concourse.web.auth.oauth.enabled }}
+            {{- if .Values.concourse.web.auth.oauth.displayName }}
+            - name: CONCOURSE_OAUTH_DISPLAY_NAME
+              value: {{ .Values.concourse.web.auth.oauth.displayName }}
+            {{- end }}
+            {{- if .Values.concourse.web.auth.oauth.clientId }}
```

Comment:
In case we don't use helm to create secrets, we can't use clientId/clientSecret in presence guards, we can safely assume that they exist in the secret either way, so I don think these presence guards are needed